### PR TITLE
build ci:add MSVC windows native ci workflow

### PR DESCRIPTION
**Pull request by xuxin930**
## Summary

Added CI verification for Windows MSVC compilation to prevent compilation blocks caused by platform issues.

## Impact

CI Checks workflow

## Testing

```cmake
# in windows native
cmake -B vs2022 -DBOARD_CONFIG=sim/windows -G"Visual Studio 17 2022" -A Win32
cmake --build vs2022
```

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



I need the content of the Pull Request to assess if it meets the requirements. Please provide the following:

* **Link to the Pull Request:** This is essential for me to review the proposed changes.
* **Description of the change:** If the PR description is not comprehensive, provide a brief explanation of what the PR aims to achieve.

Once I have this information, I can evaluate the PR against the NuttX requirements and give you a concise answer.

**Comment by lupyuen** [comment]
Sorry the workflow seems to be disabled, due to the security policy: https://github.com/apache/nuttx/actions/runs/11228519615
```text
jwlawson/actions-setup-cmake@v1 is not allowed to be used in apache/nuttx. Actions in this workflow must be: within a repository that belongs to your Enterprise account, created by GitHub, verified in the GitHub Marketplace, or matching the following:
```

**Comment by xuxin930** [comment]
> Sorry the workflow seems to be disabled, due to the security policy: https://github.com/apache/nuttx/actions/runs/11228519615
> 
> ```
> jwlawson/actions-setup-cmake@v1 is not allowed to be used in apache/nuttx. Actions in this workflow must be: within a repository that belongs to your Enterprise account, created by GitHub, verified in the GitHub Marketplace, or matching the following:
> ```

@lupyuen thanks, I tried to modify it, it seems that the windows image comes with cmake tool,
I plan to successfully complete the windows native compilation first, and then integrate your enhancements
https://github.com/apache/nuttx/issues/13775

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -331,3 +331,45 @@ jobs:
...
+
+  # Build with MSVC in Windows native
+  msvc:
+    needs: Fetch-Source
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+      # Set up Python environment and install kconfiglib
+      - name: Set up Python and install kconfiglib
+        uses: actions/setup-python@v2
```
[the remark this answers is not in the document]
Comment (xuxin930) [reply .github/workflows/build.yml]: > uses: actions/setup-python@v5

Done

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -331,3 +331,45 @@ jobs:
...
+        with:
+          python-version: '3.x'  # specify the desired Python version
```
[the remark this answers is not in the document]
Comment (xuxin930) [reply .github/workflows/build.yml]: > python-version: '3.10'

Done

**Comment by xiaoxiang781216** [comment]
@xuxin930 is it ready for merging?

**Comment by xuxin930** [comment]
> @xuxin930 is it ready for merging?

@xiaoxiang781216 Not yet,I need to integrate with labels

**Comment by xuxin930** [comment]
hi @lupyuen how can i integrate labels with this MSVC ci, for example, if we only build sim/windows here, should the changes only affect the sim change?

**Comment by lupyuen** [comment]
@xuxin930 Wonder if we can extend the Build Rules `arch.yml` + Build Workflow `build.yml`, which are described here: https://github.com/apache/nuttx/issues/13775

**Review by xuxin930 — COMMENTED** [review]
@lupyuen Thank you very much.
because MSVC cannot use the cibuild.sh script, we do not need to specify the corresponding board build list. Here I use a phony placeholder to only determine whether to skip.
Please help review whether it is correct

**Comment by xuxin930** [comment]
hi @hujun260 
Can you help me look at the MSVC issues caused by this PR? https://github.com/apache/nuttx/pull/13886

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -331,3 +331,54 @@ jobs:
           name: msys2-${{matrix.boards}}-builds
           path: buildartifacts/
         continue-on-error: true
+
+  # Select the msvc Builds based on PR Arch Label
+  msvc-Arch:
+    uses: apache/nuttx/.github/workflows/arch.yml@master
+    needs: Fetch-Source
+    with:
+      os: Linux
```
Comment (lupyuen) [inline .github/workflows/build.yml:340]: Let's change "Linux" to "msvc", so (in future) we can run special Build Rules in `arch.yml`
```suggestion
      os: msvc
```

**Review by lupyuen — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (xuxin930) [reply .github/workflows/build.yml]: thanks,Done

**Comment by xiaoxiang781216** [comment]
need fix msvc build error before merging.

File: arch/sim/src/sim/CMakeLists.txt [hunk 4]
```diff
@@ -176,7 +176,7 @@ if(CONFIG_SMP)
   list(APPEND SRCS sim_smpsignal.c sim_cpuidlestack.c)
 endif()
 
-if(CONFIG_ARCH_HAVE_MULTICPU)
+if(CONFIG_ARCH_HAVE_MULTICPU AND NOT WIN32)
```
Comment (xiaoxiang781216) [inline arch/sim/src/sim/CMakeLists.txt:179]: remove don't need

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**